### PR TITLE
lint yml added

### PR DESCRIPTION
**Pull request by eiozalp**

### CODE REVIEW

File: types.ts [hunk 1]
```diff
@@ -383,13 +393,6 @@ export type yAccessorFunc = (d: any, i?: number) => number;
 
 export type Domain = number | Date | undefined;
 
-export interface VoronoiProps {
```
Comment (eiozalp) [inline types.ts:386]: @hi2dmitri FYI  there were two Voronoi props so I deleted this one and added margin to the other one.

File: .eslintrc.json [hunk 2]
```diff
@@ -19,8 +19,9 @@
     "@typescript-eslint/no-unsafe-member-access": 0,
     "@typescript-eslint/no-unsafe-return": 0,
     "@typescript-eslint/no-unsafe-call": 0,
-    "react/display-name": 0
-
+    "react/display-name": 0,
+    "react-hooks/exhaustive-deps": "off",
+    "react-hooks/rules-of-hooks": "off"
```
Comment (robcrock) [inline .eslintrc.json:24]: I turned the rules-of-hooks on and ran lint with no errors. Why do you think we want this turned off?

File: src/charts/AreaChart/AreaChart.tsx [hunk 3]
```diff
@@ -214,7 +213,7 @@ export default function AreaChart({
   // Initialize event listeners and create interaction behavior
   // ********************
 
-  const [tooltip, setTooltip] = useState<false | any>(false);
+  const [tooltip, setTooltip] = useState<false | toolTipState>(false);
```
Comment (robcrock) [inline src/charts/AreaChart/AreaChart.tsx:216]: Good one 👍

File: src/charts/PieChart/PieChart.tsx [hunk 4]
```diff
@@ -136,7 +136,7 @@ export default function PieChart({
   // ********************
 
   const textTranform = (d: any) => {
-    const [x, y] = arcGenerator.centroid(d);
+    const [x, y]: number[] = arcGenerator.centroid(d);
```
Comment (robcrock) [inline src/charts/PieChart/PieChart.tsx:139]: This is great!

File: src/charts/PieChart/PieChart.tsx [hunk 5]
```diff
@@ -235,7 +235,7 @@ export default function PieChart({
       <svg width={'100%'} height={'100%'}>
         <g transform={translate} data-testid="pie-chart">
           {pie.map((d: any, i: number) => (
-            <g key={'g' + i}>
+            <g key={`g + ${i}`}>
```
Comment (robcrock) [inline src/charts/PieChart/PieChart.tsx:238]: Also cool to see. Much more clear.

File: types.ts [hunk 6]
```diff
@@ -4,6 +4,15 @@ export interface Data {
   [key: string]: any;
 }
 
+export interface toolTipState {
```
Comment (robcrock) [inline types.ts:7]: Love seeing this here. Thank you for creating this type 😍

**Review by robcrock — APPROVED** [review]

**Pull request merged**